src/robust.jl

```diff
@@ -66,14 +66,14 @@ end
 """
     winsor(x; prop=0.0, count=0)
 
-Return a copy of `x` with either `count` or proportion `prop` of the lowest
-elements of `x` replaced with the next-lowest, and an equal number of the
-highest elements replaced with the previous-highest.  To compute the Winsorized
-mean of `x` use `mean(winsor(x))`.
+Return a generator of all elements of `x` that replaces either `count` or
```

Comment:
Maybe let's just say "an iterator", so that in the future we can return a more specific type than an iterator. For example we could return a special `AbstractVector` type. Same for `trim` (even if it cannot be a vector since computing indices is O(N)).